aptos-move/move-examples/post_mint_reveal_nft/sources/minting.move

```diff
@@ -0,0 +1,947 @@
+        let token_config = borrow_global_mut<CollectionConfig>(@post_mint_reveal_nft);
+
+        assert!(vector::length(&token_uris) + vector::length(&token_config.token_uri) <= token_config.collection_maximum || token_config.collection_maximum == 0, error::invalid_argument(EEXCEEDS_COLLECTION_MAXIMUM));
+        vector::append(&mut token_config.token_uri, token_uris);
+        vector::append(&mut token_config.property_keys, property_keys);
+        vector::append(&mut token_config.property_values, property_values);
+        vector::append(&mut token_config.property_types, property_types);
+    }
+
+    /// Mint source certificate.
+    public entry fun mint_source_certificate(
+        nft_claimer: &signer,
+        amount: u64
+    ) acquires NFTMintConfig, PublicMintConfig, WhitelistMintConfig, CollectionConfig, SourceToken {
+        assert!(exists<CollectionConfig>(@post_mint_reveal_nft) && exists<WhitelistMintConfig>(@post_mint_reveal_nft) && exists<PublicMintConfig>(@post_mint_reveal_nft), error::permission_denied(ECONFIG_NOT_INITIALIZED));
+
+        let whitelist_mint_config = borrow_global_mut<WhitelistMintConfig>(@post_mint_reveal_nft);
+        let public_mint_config = borrow_global<PublicMintConfig>(@post_mint_reveal_nft);
+
+        let now = timestamp::now_seconds();
+        assert!((now > whitelist_mint_config.whitelist_minting_start_time && now < whitelist_mint_config.whitelist_minting_end_time) || (now > public_mint_config.public_minting_start_time && now < public_mint_config.public_minting_end_time), error::permission_denied(EMINTING_IS_NOT_ENABLED));
+
+        let nft_mint_config = borrow_global<NFTMintConfig>(@post_mint_reveal_nft);
+        let collection_config = borrow_global_mut<CollectionConfig>(@post_mint_reveal_nft);
+        let resource_signer = create_signer_with_capability(&nft_mint_config.signer_cap);
+
+        // Set up a source token if it doesn't exist yet.
+        if (!exists<SourceToken>(@post_mint_reveal_nft)) {
+            // This is the collection that contains certificate tokens, which will allow users to exchange to an NFT in the destination collection after reveal time.
+            let source_collection_name = collection_config.collection_name;
+            string::append_utf8(&mut source_collection_name, b" - source collection");
+            create_collection(&resource_signer, source_collection_name, collection_config.collection_description, utf8(b""), collection_config.collection_maximum, collection_config.collection_mutate_config);
+
+            let source_token_data_id = create_tokendata(
+                &resource_signer,
+                source_collection_name,
+                // token name
+                utf8(b""),
+                // token description
+                utf8(b""),
+                // token maximum is equal to the number of tokens in the destination collection
+                vector::length(&collection_config.token_uri),
```

Comment:
why?

Reply:
because we don't want to hand out more source certificate tokens than the number of tokens in the destination collection?